### PR TITLE
Extend example of binding parameter estimation to include parameter transformation

### PR DESCRIPTION
**Pull request by ronald-jaepel**

### CODE REVIEW

**Comment by ronald-jaepel** [comment]
I've extended the documentation to also include a section on how to use variable dependencies with kEq and kKin. IMO this PR is ready for Review and then merging.

File: docs/source/user_guide/optimization/variable_dependencies.md [hunk 1]
```diff
@@ -61,3 +58,67 @@ optimization_problem.add_variable_dependency('var_2', ['var_0', 'var_1'], transf
 ```
 
 Note that generally bounds and linear constraints can still be specified independently for all variables.
+
+## Adsorption rates example
+
+For instance, consider an adsorption proces with an adsorption rate $k_a$ and a desorption rate $k_d$. Both
```
Comment (schmoelder) [inline docs/source/user_guide/optimization/variable_dependencies.md:64]: Please use one line per sentence.

File: examples/characterize_chromatographic_system/binding_model_parameters.md [hunk 2]
```diff
@@ -264,7 +300,12 @@ if __name__ == '__main__':
         )
 
 
-    optimization_problem.add_callback(callback, requires=[simulator])
+    optimization_problem.add_callback(callback, requires=[simulator], callbacks_dir="tmp")
```
Comment (schmoelder) [inline examples/characterize_chromatographic_system/binding_model_parameters.md:303]: is the callbacks_dir required here?

**Review by schmoelder — CHANGES_REQUESTED** [review]
Thanks for your input. I have just some minor comments. Otherwise, LGTM! :)

File: examples/characterize_chromatographic_system/binding_model_parameters.md (hunk 2 above)
Comment (ronald-jaepel) [reply examples/characterize_chromatographic_system/binding_model_parameters.md:303]: Without that, the optimization_problem.evaluate_callbacks(ind) in line 308 throws an error.

**Pull request merged**